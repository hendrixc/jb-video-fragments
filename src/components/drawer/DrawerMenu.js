import React, { Component } from 'react';
import { Icon } from '@rmwc/icon';
import {
  Drawer,
  DrawerHeader,
  DrawerContent,
  DrawerTitle,
  DrawerSubtitle,
} from '@rmwc/drawer';
import {
  List,
  ListItem,
} from '@rmwc/list';

import '@material/drawer/dist/mdc.drawer.css';
import '@rmwc/icon/icon.css';
import './drawerMenu.css';
import colors from '../../style/colors';
import I18n from '../../i18n';
import ShortcutCard from '../shortcuts/ShortcutCard';

const iconStyle = { color: colors.secondary300, paddingRight: 10 };
class DrawerMenu extends Component {
  static getDrawerHeader() {
    return (
      <DrawerHeader>
        <DrawerTitle>{I18n.t('drawer.header')}</DrawerTitle>
        <DrawerSubtitle>{I18n.t('drawer.subtitle')}</DrawerSubtitle>
      </DrawerHeader>
    );
  }


  static getList() {
    return (
      <List>
        {DrawerMenu.getListItem('favorite', I18n.t('drawer.favorites'))}
        {DrawerMenu.getListItem('search', I18n.t('drawer.search'))}
        {DrawerMenu.getListItem('lock', I18n.t('drawer.account'))}
      </List>
    );
  }

  static getListItem(icon, text) {
    return (
      <ListItem>
        <div className="item">
          <Icon
            icon={icon}
            style={iconStyle}
          />
          <h4>{text}</h4>
        </div>
      </ListItem>
    );
  }

  render() {
    return (
      <div>
        <Drawer>
          {DrawerMenu.getDrawerHeader()}
          <DrawerContent>
            {DrawerMenu.getList(iconStyle)}
          </DrawerContent>
        </Drawer>
        <ShortcutCard />
      </div>
    );
  }
}

export default DrawerMenu;
