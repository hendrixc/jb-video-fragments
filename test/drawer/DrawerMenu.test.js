import React from 'react';
import renderer from 'react-test-renderer';
import DrawerMenu from '../../src/components/drawer/DrawerMenu';

it('renders an home component into the DrawerMenu as snapshot', () => {
  const TextInputComponent = renderer.create(<DrawerMenu />).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});
