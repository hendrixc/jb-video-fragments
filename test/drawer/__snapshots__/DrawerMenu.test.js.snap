// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an home component into the DrawerMenu as snapshot 1`] = `
<div>
  <aside
    className="mdc-drawer"
  >
    <div
      className="mdc-drawer__header"
    >
      <div
        className="mdc-drawer__title"
      >
        Anonymous User!!
      </div>
      <div
        className="mdc-drawer__subtitle"
      >
        Welcome!!
      </div>
    </div>
    <div
      className="mdc-drawer__content"
    >
      <div
        className="mdc-list"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
      >
        <div
          className="mdc-list-item"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="item"
          >
            <i
              className="rmwc-icon material-icons"
              style={
                Object {
                  "color": "#F06292",
                  "paddingRight": 10,
                }
              }
            >
              favorite
            </i>
            <h4>
              Favorites
            </h4>
          </div>
        </div>
        <div
          className="mdc-list-item"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="item"
          >
            <i
              className="rmwc-icon material-icons"
              style={
                Object {
                  "color": "#F06292",
                  "paddingRight": 10,
                }
              }
            >
              search
            </i>
            <h4>
              Search
            </h4>
          </div>
        </div>
        <div
          className="mdc-list-item"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="item"
          >
            <i
              className="rmwc-icon material-icons"
              style={
                Object {
                  "color": "#F06292",
                  "paddingRight": 10,
                }
              }
            >
              lock
            </i>
            <h4>
              Account
            </h4>
          </div>
        </div>
      </div>
    </div>
  </aside>
</div>
`;
